src/main/java/edu/ohio/ais/rundeck/HttpBuilder.java

```diff
@@ -107,8 +109,15 @@ public boolean isTrusted(X509Certificate[] x509Certificates, String s) throws Ce
             httpClientBuilder.setSSLHostnameVerifier(new NoopHostnameVerifier());
             httpClientBuilder.setSSLContext(sslContextBuilder.build());
         }
-        if(options.containsKey("proxySettings") && Boolean.parseBoolean(options.get("proxySettings").toString())){
-            HttpHost proxy = new HttpHost(options.get("proxyIP").toString(), Integer.valueOf((String)options.get("proxyPort")), "http");
+        if(getBooleanOption(options, "useSystemProxySettings", false) && !getBooleanOption(options, "proxySettings", false)) {
+            log.log(5, "Using proxy settings set on system");
+            HttpHost proxy = new HttpHost(System.getProperty("http.proxyHost"), Integer.parseInt(System.getProperty("http.proxyPort")), "http");
+            httpClientBuilder.setProxy(proxy);
+        }
+        if(getBooleanOption(options, "proxySettings", false)){
+            log.log(5, "proxy IP set in job: " + options.get("proxyIP").toString());
```

Comment:
could this lead to a NPE? perhaps it should be validated and throw a "StepException" 🤔